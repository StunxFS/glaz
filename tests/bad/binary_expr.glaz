fn main() {
  _ = true + false;
  _ = true and (22 or "");
}
