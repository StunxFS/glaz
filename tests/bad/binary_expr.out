tests/bad/binary_expr.glaz:2:8: error: booleans only support `==`, `!=`, `and`, `or`, `&` and `|`
     | 
   2 |   _ = true + false;
     |       ^^^^^^^^^^^^
tests/bad/binary_expr.glaz:3:18: error: non-boolean expression in left operand for `or`
     | 
   3 |   _ = true and (22 or "");
     |                 ^^
tests/bad/binary_expr.glaz:3:24: error: non-boolean expression in right operand for `or`
     | 
   3 |   _ = true and (22 or "");
     |                       ^^
tests/bad/binary_expr.glaz:3:24: error: expected type `untyped_int`, found `str`
     | 
   3 |   _ = true and (22 or "");
     |                       ^^
tests/bad/binary_expr.glaz:3:17: error: non-boolean expression in right operand for `and`
     | 
   3 |   _ = true and (22 or "");
     |                ^^^^^^^^^^
tests/bad/binary_expr.glaz:3:17: error: expected type `bool`, found `untyped_int`
     | 
   3 |   _ = true and (22 or "");
     |                ^^^^^^^^^^
glaz error: could not compile `binary_expr`, aborting due to 6 previous errors.
